lib/dynamoid/adapter_plugin/aws_sdk_v2.rb

```diff
@@ -539,12 +546,20 @@ def scan(table_name, scan_hash, select_opts = {})
 
         Enumerator.new do |y|
           # Batch loop, pulls multiple requests until done using the start_key
+          # by paging in by batch size rather than limit size
+          eval_count = 0
           loop do
             results = client.scan(request)
-
             results.data[:items].each { |row| y << result_item_to_hash(row) }
 
-            if((lk = results[:last_evaluated_key]) && batch)
+            # If limit is set then keep track and stop requesting once
+            # completed. This is eval_limit so we compare to scanned_count
+            # rather than resulting data
+            eval_count += results.scanned_count
```

Comment:
@andrykonchin @richardhsu 
I like the idea of splitting the limits and having separate names.  `scan_limit` is very clear for that concept.

I like the idea of *not* using the familiar `limit` because then people won't assume an incorrect meaning if they don't read the documentation.  Since you pay for what you use this matters.  People could assume `limit` will prevent reading more than `limit` records, or that it will read as much as needed until it finds `limit` records.

As such I like using something differentiated from the standard `limit` token.  I agree that `eval_limit` is strange, because of the *eval is evil* mantra, and there is no Ruby eval happening here.

Other potential ideas:
• `result_limit`
• `record_limit`